File: app/src/main/java/me/liuyun/bjutlgn/db/FlowManager.java
package me.liuyun.bjutlgn.db;

import android.content.Context;
import android.util.Log;

import java.util.List;

import me.liuyun.bjutlgn.entity.Flow;

public class FlowManager {
    private static final String TAG = "FlowManager";

    private DBHelper dbHelper;

    public FlowManager(Context context) {
        this.dbHelper = new DBHelper(context);
    }

    public boolean insertFlow(long timestamp, int flow) {
        try {
            dbHelper.getFlowDao().create(new Flow(0, timestamp, flow));
            return true;
        } catch (Exception e) {
            Log.e(TAG, "insertFlow", e);
        }
        return false;
    }

    public List<Flow> getAllFlow() {
        try {
            return dbHelper.getFlowDao().queryForAll();
        } catch (Exception e) {
            Log.e(TAG, "getAllFlow", e);
        }
        return null;
    }

    public boolean clearFlow() {
        try {
            dbHelper.getFlowDao().deleteBuilder().delete();
            return true;
        } catch (Exception e) {
            Log.e(TAG, "clearFlow", e);
        }
        return false;
    }
}

File: app/src/main/kotlin/me/liuyun/bjutlgn/db/FlowManager.kt
package me.liuyun.bjutlgn.db

import android.content.Context
import android.util.Log
import com.j256.ormlite.dao.Dao
import me.liuyun.bjutlgn.entity.Flow

class FlowManager(context: Context) {

    private val dbHelper: DBHelper = DBHelper(context)

    fun transaction(body: (Dao<Flow, Int>) -> Int): Int {
        try {
            return  body(dbHelper.flowDao)
        } catch (e: Exception) {
            Log.e(TAG, body.toString(), e)
        }
        return 0
    }

    fun <T> transaction(body: (Dao<Flow, Int>) -> MutableList<T>): MutableList<T> {
        try {
            return body(dbHelper.flowDao)
        } catch (e: Exception) {
            Log.e(TAG, body.toString(), e)
        }
        return mutableListOf()
    }

    fun insertFlow(timestamp: Long, flow: Int): Int {
        return transaction { it.create(Flow(0, timestamp, flow)) }
    }

    fun clearFlow(): Int {
        return transaction { it.deleteBuilder().delete() }
    }

    fun allFlow(): MutableList<Flow> {
        return transaction<Flow> { it.queryForAll() }
    }

    val TAG: String = this.javaClass.name
}
